xl/event.py

```diff
@@ -297,18 +351,21 @@ def add_callback(self, function, type, obj, args, kwargs):
             remove the callback.
         """
         with self.lock:
+            cb = Callback(function, time.time(), args, kwargs)
+            
             # add the specified categories if needed.
-            if not self.callbacks.has_key(type):
-                self.callbacks[type] = weakref.WeakKeyDictionary()
-            if obj is None:
-                obj = _NONE
-            try:
-                callbacks = self.callbacks[type][obj]
-            except KeyError:
-                callbacks = self.callbacks[type][obj] = []
-
-            # add the actual callback
-            callbacks.append(Callback(function, time.time(), args, kwargs))
+            for cbs in [self.callbacks, self.all_callbacks, self.ui_callbacks]:
```

Comment:
This adds `cb` to all three variables; you just want it on `all_callbacks` and one of the others.